## Rubric

Instructions: DELETE this cell before you submit via a `git push` to your repo before deadline. This cell is for your reference only and is not needed in your report. 

 Scoring: Out of 10 points

- Each Developing  => -2 pts
- Each Unsatisfactory/Missing => -4 pts
  - until the score is 0

If students address the detailed feedback in a future checkpoint they will earn these points back






|                                  | **Unsatisfactory**                                                                                                                                                                                                                                                                                                                        | **Developing**                                                                                                                                                                                                       | **Proficient**                                                                                                                                                                                            | **Excellent**                                                                                                                                                                            |
|----------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **EDA relevance**                | EDA is mostly neither relevant to the question nor helpful in figuring out how to address the question. Or the EDA does address the question, but many obviously relevant variables / analyses / figures were not included. | EDA is partly irrelevant/unhelpful. EDA missed one or two obvioulsy relevant analysis (distributions of single variables or relationships between variables) | EDA includes the obviously relevant / helpful variables in addressing the question.                                                              | Thorough EDA fully explored the dataset                                                                                                                 |
| **EDA analysis and description** | Many of the analyses are poor choices (e.g., using means instead of medians for obviously skewed data), or are poorly described in the text, or do not aid understanding the data                                                                                                                                                     | Some of the analyses are poor choices, or are poorly described in the text, or do not aid understanding the data                                                                                                 | All analyses are correct choices. Only one or two have minor issues in the text descriptions supporting them. Mostly they fit well with other elements of the EDA and support understanding the data  | All analyses are correct choices with clear text descriptions supporting them. The figures fit well with the other elements of the EDA, producing a clear understanding of the data. |
| **EDA figures**                  | Many of the figures are poor plot choices (e.g., using a bar plot to represent a time series where it would be better to use a line plot) or have poor aesthetics (including colormap, data point shape/color, axis labels, titles, annotations, text legibility) or do not aid understanding the data                                | Some of the figures are poor plot choices or have poor aesthetics. Some figures do not aid understanding the data                                                                                                | All figures are correct plot choices. Only one or two have minor questionable aesthetic choices. The figures mostly fit well with the other elements of the EDA and support understanding the data    | All figures are correct plot choices with beautiful aesthetics. The figures fit well with the other elements of the EDA, producing a clear understanding of the data.                |





# COGS 108 - EDA Checkpoint

## Authors

Instructions: REPLACE the contents of this cell with your team list and their contributions. Note that this will change over the course of the checkpoints

This is a modified [CRediT taxonomy of contributions](https://credit.niso.org). For each group member please list how they contributed to this project using these terms:
> Analysis, Background research, Conceptualization, Data curation, Experimental investigation, Methodology, Project administration, Software, Visualization, Writing – original draft, Writing – review & editing

Example team list and credits:
- Alice Anderson: Conceptualization, Data curation, Methodology, Writing - original draft
- Bob Barker:  Analysis, Software, Visualization
- Charlie Chang: Project administration, Software, Writing - review & editing
- Dani Delgado: Analysis, Background research, Visualization, Writing - original draft

# Research Question

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback



## Background and Prior Work

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback

# Hypothesis


Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback

## Data

### Data overview

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your data checkpoint feedback


In [37]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [38]:
# Setup code -- Run only once after cloning!!! 
#
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/airline-safety/airline-safety.csv', 'filename':'airline-safety.csv'},
    { 'url': 'https://raw.githubusercontent.com/GrzegorzPus/NBA_analysis/9bf80ea01119e20c5dc04983b8620981d88b7b6a/Data/draft-data-20-years.csv', 'filename': 'draft-data-20-years.csv'},
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/bad-drivers/bad-drivers.csv', 'filename':'bad-drivers.csv'}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

Overall Download Progress:  33%|███▎      | 1/3 [00:00<00:00,  6.39it/s]   

Successfully downloaded: airline-safety.csv



Overall Download Progress:  67%|██████▋   | 2/3 [00:00<00:00,  5.95it/s]       

Successfully downloaded: draft-data-20-years.csv



Overall Download Progress: 100%|██████████| 3/3 [00:00<00:00,  6.26it/s]

Successfully downloaded: bad-drivers.csv


### Dataset #1 

Instructions: REPLACE the contents of this cell and the one below with your work, including any updates to recover points lost in your data checkpoint feedback

In [39]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION 

### Dataset #2
### 20 Years of NBA Draft Data from Kaggle

For Dataset #2, I am using NBA Draft data from Kaggle. This data shows players' draft year, college they played in, which team they were drafted to, and their overall draft rankings in the last 20 years. This data can be compared to dataset 3 to see their performance pre and post NBA. This dataset mainly focuses on their overall pick in the draft, but variables like position, draft year (who they were drafted with) can effect the result.

### A: Loading Data Set

In [48]:
raw_path = 'data/00-raw/draft-data-20-years.csv'
df_draft = pd.read_csv(raw_path)

print("Dataset Shape:", df_draft.shape)
print("\nColumns:\n", df_draft.columns)

df_draft.head()

Dataset Shape: (1868, 26)

Columns:
 Index(['Unnamed: 0', 'Rk', 'Pk', 'Tm', 'Player', 'College', 'Yrs', 'G',
       'TOTMP', 'TOTPTS', 'TOTTRB', 'TOTAST', 'FG%', '3P%', 'FT%', 'WS',
       'WS/48', 'BPM', 'VORP', 'DraftYr', 'MPG', 'PPG', 'RPG', 'APG',
       'playerurl', 'DraftYear'],
      dtype='object')


,Unnamed: 0,Rk,Pk,Tm,Player,College,Yrs,G,TOTMP,TOTPTS,...,WS/48,BPM,VORP,DraftYr,MPG,PPG,RPG,APG,playerurl,DraftYear
0,1,1,1,BRK,Derrick Coleman,Syracuse,15.0,781.0,25903.0,12884.0,...,0.119,1.4,22.3,1990,33.2,16.5,9.3,2.5,https://www.sports-reference.com/cbb/players/d...,1990
1,2,2,2,OKC,Gary Payton,Oregon State,17.0,1335.0,47117.0,21813.0,...,0.148,3.3,62.5,1990,35.3,16.3,3.9,6.7,https://www.sports-reference.com/cbb/players/g...,1990
2,3,3,3,DEN,Mahmoud Abdul-Rauf,LSU,9.0,586.0,15628.0,8553.0,...,0.077,-0.8,4.5,1990,26.7,14.6,1.9,3.5,https://www.sports-reference.com/cbb/players/m...,1990
3,4,4,4,ORL,Dennis Scott,Georgia Tech,10.0,629.0,17983.0,8094.0,...,0.089,0.2,9.9,1990,28.6,12.9,2.8,2.1,https://www.sports-reference.com/cbb/players/d...,1990
4,5,5,5,CHA,Kendall Gill,Illinois,15.0,966.0,29481.0,12914.0,...,0.078,0.1,15.8,1990,30.5,13.4,4.1,3.0,https://www.sports-reference.com/cbb/players/k...,1990


### B: Demonstrate tidiness / Checking for missing values 

In [49]:
# Checking if any of the values are missing 
missing_data = df_draft.isnull().sum()
print("Missing values per column:\n", missing_data)
display(df_draft.head())

Missing values per column:
 Unnamed: 0      0
Rk              0
Pk              0
Tm              0
Player          0
College       335
Yrs           247
G             247
TOTMP         247
TOTPTS        247
TOTTRB        247
TOTAST        247
FG%           252
3P%           368
FT%           284
WS            247
WS/48         248
BPM           248
VORP          247
DraftYr         0
MPG           247
PPG           247
RPG           247
APG           247
playerurl     335
DraftYear       0
dtype: int64


,Unnamed: 0,Rk,Pk,Tm,Player,College,Yrs,G,TOTMP,TOTPTS,...,WS/48,BPM,VORP,DraftYr,MPG,PPG,RPG,APG,playerurl,DraftYear
0,1,1,1,BRK,Derrick Coleman,Syracuse,15.0,781.0,25903.0,12884.0,...,0.119,1.4,22.3,1990,33.2,16.5,9.3,2.5,https://www.sports-reference.com/cbb/players/d...,1990
1,2,2,2,OKC,Gary Payton,Oregon State,17.0,1335.0,47117.0,21813.0,...,0.148,3.3,62.5,1990,35.3,16.3,3.9,6.7,https://www.sports-reference.com/cbb/players/g...,1990
2,3,3,3,DEN,Mahmoud Abdul-Rauf,LSU,9.0,586.0,15628.0,8553.0,...,0.077,-0.8,4.5,1990,26.7,14.6,1.9,3.5,https://www.sports-reference.com/cbb/players/m...,1990
3,4,4,4,ORL,Dennis Scott,Georgia Tech,10.0,629.0,17983.0,8094.0,...,0.089,0.2,9.9,1990,28.6,12.9,2.8,2.1,https://www.sports-reference.com/cbb/players/d...,1990
4,5,5,5,CHA,Kendall Gill,Illinois,15.0,966.0,29481.0,12914.0,...,0.078,0.1,15.8,1990,30.5,13.4,4.1,3.0,https://www.sports-reference.com/cbb/players/k...,1990


### C: Demonstrate the size of the dataset

In [50]:
print("Rows:", df_draft.shape[0])
print("Columns:", df_draft.shape[1])

Rows: 1868
Columns: 26


### D: Clean the Data

In [51]:
# Keeping important variables only to be used as columns (total of 6)
columns_to_keep = ['Player', 'DraftYr', 'Rk', 'Pk', 'College', 'Tm', 'WS', 'G']
df_draft_clean = df_draft[columns_to_keep].dropna(subset=['Pk'])
print("Final cleaned dataset shape:", df_draft_clean.shape)

Final cleaned dataset shape: (1868, 8)


### E: Detect Outliers or Suspicious Entries

In [52]:
# Since draft pick # range from 1 - 60, checking if any of the numbers are not in this range.
invalid_picks = df_draft_clean[(df_draft_clean['Pk'] < 1) | (df_draft_clean['Pk'] > 60)]
print(f"Invalid # draft picks found: {len(invalid_picks)}")

Invalid # draft picks found: 0


### F. Save Cleaned Dataset

In [53]:
processed_path_draft = 'data/02-processed/dataset_2_draft_clean.csv'
df_draft_clean.to_csv(processed_path_draft, index=False)

print("Cleaned dataset saved.")
df_draft_clean.head()

Cleaned dataset saved.


,Player,DraftYr,Rk,Pk,College,Tm,WS,G
0,Derrick Coleman,1990,1,1,Syracuse,BRK,64.3,781.0
1,Gary Payton,1990,2,2,Oregon State,OKC,145.5,1335.0
2,Mahmoud Abdul-Rauf,1990,3,3,LSU,DEN,25.2,586.0
3,Dennis Scott,1990,4,4,Georgia Tech,ORL,33.4,629.0
4,Kendall Gill,1990,5,5,Illinois,CHA,47.8,966.0


## Summary Statistic


In [54]:
stats_columns = ['Pk', 'WS', 'G']
summary_stats = df_draft_clean[stats_columns].describe()
summary_stats.loc['range'] = summary_stats.loc['max'] - summary_stats.loc['min']
print("Summary Statistics for NBA Draft Dataset (2000-2021)")
display(summary_stats.round(2))

Summary Statistics for NBA Draft Dataset (2000-2021)


,Pk,WS,G
count,1868.00,1621.00,1621.00
mean,29.76,17.58,342.48
std,16.95,27.86,322.95
min,1.00,-1.90,1.00
25%,15.00,0.40,69.00
50%,30.00,4.90,230.00
75%,44.00,23.80,575.00
max,60.00,248.30,1541.00
range,59.00,250.20,1540.00


This shows that:
- Pk mean is 29.76 and 50% (median) is 30, which means that the dataset is perfectly balanced since draft number range from 1 to 60.
- G is longevity, which shows that 25% of the drafted players do not make it through 69.0 games. This means that the draft failure rate is high. (should look at players who hit the 50% 230 games to compare to data from dataset 1)
- WS is Win shares, and mean being 17.58 and median 4.90 means that the data is right-skewed. Mean being 3 times higher than the median shows that most players in the NBA draft have low career impact and select few superstars bring the average up.

### Dataset #3
### NBA Players Stats since 1950s from Kaggle
 
For Dataset #3, I am using a dataset from Kaggle called NBA player stats. This dataset contains
individual statistics for each player across 67 seasons. There are many useful column variables to
work with for measuring a player’s career success, such as win shares, win shares per 48 games,
true shooting percentage, and more. However, since this dataset contains players from before the 2000s and
many of these statistics didn’t exist before then, we need to consider filtering these players out of
our data (dataset #2 also only contains players beyond 2000s, so we should filter out players before then). Additionally, since the dataset contains repeated entries of players due to it capturing the stats of players for every season, we need to sum up or average the stats of interest for their total career statistics. 

There many column variables in this dataset, including win shares, true shooting percentage, age, block percentage, free throw rate, etc. However, when measuring a player's career success, there are a few columns that can be major indicators. The columns we are most interested in are win shares, true shooting percentage, total career points, total games played, and total career minutes. 

The winshares are a cumulative statistic that estimates the exact number of team wins contributed by a player due to both their offense and defense. Essentially, they represent the overall impact on team success made by a player. The true shooting percentage measures how effectively a player scores, accounting for 2-pointers, 3-pointers and free throws in a single metric. This statistic represents the offensive efficiency of a player. The total points are the total points scored by the player throughout their career, which represents their raw offensive output. The total games represent the longetivity and durability by a player because it measures how long a player remained healthy throughout their career. The total minutes is how much time the player spent actually playing. which is why it represents the level of trust the player's coach has with the player and how big of a role they played for their teams. When we combine these statistics together, we are evaluating the overall success impact, offensive efficiency, volume of success, reliability, and longevity of a player, which effectively captures the player's career success. 

### A: Loading Data Set


In [41]:
import pandas as pd
import numpy as np
from scipy import stats
import seaborn as sns
import matplotlib.pyplot as plt
import os
# load both career and draft data to prepare for merge
raw_path = 'data/00-raw/draft-data-20-years.csv'
df_combine = pd.read_csv(raw_path)
df_career = pd.read_csv('data/00-raw/Seasons_Stats.csv')
# merge into dataframe with all players with both combine and career data
df_final = pd.merge(df_combine, df_career, on='Player', how='inner')

FileNotFoundError: [Errno 2] No such file or directory: 'data/00-raw/Seasons_Stats.csv'

### B: Make the dataset tidy or demonstrate that it was already tidy

In [20]:
 # create a new dataframe with ONLY the columns we are interested in
my_columns = [
'Player', 'Draft_Year',
'G_y', 'MP', 'WS_y', 'TS%', 'PTS']
existing_cols = [c for c in my_columns if c in df_final.columns]
df_project1 = df_final[existing_cols].copy()
df_project1 = df_project1.rename(columns={
'G_y': 'Games',
'WS_y': 'Win_Shares',
'MP': 'Minutes',
'PTS': 'Points',
'TS%': 'True_Shooting'
})

NameError: name 'df_final' is not defined

In [21]:
 # Sum the volume stats and average the true shooting percentage
aggregations = {
'Games': 'sum',
'Minutes': 'sum',
'Win_Shares': 'sum',
'Points': 'sum',
'True_Shooting': 'mean'
}
# since this dataset shows each player's season stats, we want to sum all of their seasons stats into one single data for each player
df_project = df_project1.groupby('Player').agg(aggregations).reset_index()
df_project.head(10)

NameError: name 'df_project1' is not defined

### C: Demonstrate the size of the dataset

In [22]:
print(f"Dataset Dimensions: {df_project.shape}")
print(f"Rows (Players): {df_project.shape[0]}")
print(f"Columns (Variables): {df_project.shape[1]}")

NameError: name 'df_project' is not defined

### D: Investigate Missing Data

In [23]:
# Count missing values per column
missing_counts = df_project.isnull().sum()
print("\n--- Missing Data Analysis---")
print(missing_counts[missing_counts > 0]) # Only print columns with gaps

# Calculate percentage of missing data
print((missing_counts / len(df_project)) * 100)

NameError: name 'df_project' is not defined

### E: Detect Outliers or Suspicious Entries

In [24]:
numeric_cols = ['Games', 'Minutes', 'Win_Shares', 'Points', 'True_Shooting']
print("---Outlier Detection Report---")
# detect outlier using the interquartile method
for col in numeric_cols:
    
# calculate quartiles
    Q1 = df_project[col].quantile(0.25)
    Q3 = df_project[col].quantile(0.75)
    IQR = Q3- Q1
    lower_bound = Q1- 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

# Find the outliers
outliers = df_project[(df_project[col] < lower_bound) | (df_project[col] > upper_bound)]

# Print findings
print(f"\n Variable: {col}")
print(f"- Normal Range: {lower_bound:.2f} to {upper_bound:.2f}")
print(f"- Outliers Found: {len(outliers)}")

# Show the top 3 biggest outliers
if not outliers.empty:
    print(f"- Top 3 Outliers:")
    print(outliers[['Player', col]].sort_values(by=col, ascending=False).head(3).to_string(index=False))
    print("\n-----------------------------")
    
suspicious_negatives = ['Games', 'Minutes', 'Points'] #cannot be negative here, (note that win shares can be negtative if a player performs bad enough)
print("--- Suspicious Value Check (Negative Values)---\n")

# Check for the strictly positive columns
for col in suspicious_negatives:
    # Filter for values less than 0
    negative_values = df_project[df_project[col] < 0]
if len(negative_values) > 0:
    print(f"\nFound {len(negative_values)} negative values in '{col}'")
    print(negative_values[['Player', col]])
else:
    print(f"'{col}': No negative values found. (Clean)\n")
    print("------------------------------------------")

---Outlier Detection Report---


NameError: name 'df_project' is not defined

### F: Clean the Data

In [25]:
# Fill missingnumeric stats with 0
cols_to_fix= ['Games', 'Minutes', 'Win_Shares','Points', 'True_Shooting']
for col in cols_to_fix:
    df_project[col]= df_project[col].fillna(0)

# Verify cleaning
print(f"\nMissing values after cleaning: {df_project.isnull().sum().sum()}")

NameError: name 'df_project' is not defined

### G. Save Cleaned Dataset

In [26]:
processed_path_career= 'data/02-processed/dataset_3_career_clean.csv'

df_project.to_csv(processed_path_career,index=False)

print("Cleaned dataset saved.")

NameError: name 'df_project' is not defined

### Summary Statistics of Dataset #3

In [27]:
summary = df_project.describe().round(2)

print("\n--- Summary Statistics (Career Data)---")
print(summary)
print("\n")

NameError: name 'df_project' is not defined

What this means:

These summary statistics show a right-skewed distribiution in player performance, where the “average” entry represents a player who plays around 389 games and scores a total of 3828 points in their career, where as the 50% median is lower at around 1801 points. This data also shows that there is a major gap between typical players and elite players, where the top 25% of players generate an impressive 206.30 winshares and 33,643 total points, which is disproportionately greater than the rest. This is the reason why the mean is noticeably greater than the median; the elite players are skewing the avergae up. The maximum values, such as 33,643 points, align with historical NBA records which confirms the data cleaning and tidying process.

## Results

### Exploratory Data Analysis

Instructions: replace the words in this subsection with whatever words you need to setup and preview the EDA you're going to do.   

Please explicitly load the fully wrangled data you will use from `data/02-processed`.  This is a good idea rather than forcing people to re-run the data getting / wrangling cells above.  Sometimes it takes a long time to get / wrangle data compared to reloading the fixed up dataset.

Carry out whatever EDA you need to for your project in the code cells below.  Because every project will be different we can't really give you much of a template at this point. But please make sure you describe the what and why in text here as well as providing interpretation of results and context.

Please note that you should consider the use of python modules in your work.  Any code which gets called repeatedly should be modularized. So if you run the same pre-processing, analysis or visualiazation on different subsets of the data, then you should turn that into a function or class.  Put that function or class in a .py file that lives in `modules/`.  Import the module you made and use it to get your work done.  For reference see `get_raw()` which is inside `modules/get_data.py`. 



#### Section 1 of EDA - please give it a better title than this

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.

In [55]:
# Enable autoreload for our custom modules
%load_ext autoreload
%autoreload 2

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sys

# Append the modules directory to the system path
sys.path.append('./modules')
import eda_utils # Custom module we will create for repetitive tasks

# 1. Load the processed datasets
combine_df = pd.read_csv('data/02-processed/draft_combine_body_measurements.csv')
draft_df = pd.read_csv('data/02-processed/dataset_2_draft_clean.csv')
career_df = pd.read_csv('data/02-processed/dataset_3_career_clean.csv')

# Preview the successful load
print(f"Combine Data: {combine_df.shape}")
print(f"Draft Data: {draft_df.shape}")
print(f"Career Data: {career_df.shape}")

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


ModuleNotFoundError: No module named 'eda_utils'

First, we want to understand how a player's physical measurements influence their overall draft ranking (`Pk`). By merging our combine dataset with our draft dataset, we can observe if the NBA draft prioritize factors like `height_in` or `wingspan_in` when making early draft picks.

In [29]:

# Plot Height vs. Draft Pick, colored by Position
eda_utils.plot_feature_vs_target(
    df=draft_combine_merged, 
    feature_col='height_in', 
    target_col='Pk', 
    title='Player Height vs. Overall Draft Pick by Position',
    hue_col='Position'
)

# Plot Wingspan vs. Draft Pick, colored by Position
eda_utils.plot_feature_vs_target(
    df=draft_combine_merged, 
    feature_col='wingspan_in', 
    target_col='Pk', 
    title='Player Wingspan vs. Overall Draft Pick by Position',
    hue_col='Position'
)

NameError: name 'eda_utils' is not defined

The scatter plots above reveal a wide. variance and a seemingly weak relationship between physical measurements and overall draft pick. Here, 'Pk' represents how early the player was picked in the draft, so a lower number means better. Therefore, the slightly downward regression line indicates a small trend where taller and wider wingspan players are drafter earlier. However, the massive spread of data points shows that NBA draft placements are influenced by more than just physical attributes, such as skill, college performance and team needs. There is also a confounding variable which is player position. For instance, a 6'6" point guard (considered tall for a point guard) might be drafted much earlier than a 6'6" center (considered small for a center), which is why we have taken position into account as well. 

Based on the positions, we can see that there are more centers and power forwards (the positions who need bigger players) that have lower draft picks who are very tall and have wide wingspans. Around the middle, we can see many small forwards and shooting guards near the bottom, which makes sense considering that their positions are suited for 'medium' sized players. Then, around the 6 foot 2, and 75-78 inch wingspan range, we can see more point guards near the bottom. This shows that the judges in draft picks prefer players whose builds are suited for their position, and not to mention that it is rare to see point guards who are very tall, or centers that are very small, etc. 

### Physical Foundations of Long-Term NBA Success

Draft position is just an initial evaluation. The true measure of an NBA player's success can be defined by their longevity (`Games`), scoring (`Points`), and overall impact (`Win_Shares`). We will now merge our combined physical data with the career stats dataset to see if traits like `body_fat_pct` or `wingspan_in` have a statistical relationship with a player's long-term value.

In [36]:
# Merge the previously merged draft/combine dataframe with career stats
full_merged_df = pd.merge(draft_combine_merged, career_df, on='Player', how='inner')

# Drop any potential NaN values created during the merge
full_merged_df.dropna(subset=['Win_Shares', 'Games', 'wingspan_in'], inplace=True)

# Plot Wingspan against Career Win Shares
eda_utils.plot_feature_vs_target(
    df=full_merged_df, 
    feature_col='wingspan_in', 
    target_col='Win_Shares', 
    title='Wingspan vs. Career Win Shares'
)

# Correlation matrix for a holistic view of numerical features
plt.figure(figsize=(10, 8))
correlation_cols = ['height_in', 'weight_lbs', 'wingspan_in', 'body_fat_pct', 'Pk', 'Games', 'Win_Shares', 'Points']
sns.heatmap(full_merged_df[correlation_cols].corr(), annot=True, cmap='coolwarm', fmt=".2f")
plt.title("Correlation Heatmap: Combine Metrics vs. Career Success")
plt.show()

NameError: name 'draft_combine_merged' is not defined

#### Section 2 of EDA if you need it  - please give it a better title than this

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.

In [ ]:
## YOUR CODE HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION

## Ethics

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback

## Team Expectations 

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback

## Project Timeline Proposal

Instructions: Replace this with your timeline.  **PLEASE UPDATE your Timeline!** No battle plan survives contact with the enemy, so make sure we understand how your plans have changed.  Also if you have lost points on the previous checkpoint fix them